### PR TITLE
Update build script

### LINKED ISSUE #11817: Add README.md for build script
We have a good build script ( https://github.com/prettier/prettier/tree/main/scripts/build ).

It's a little complicated. It would be nice to write a friendly README.md for future maintainers.

### PR DESCRIPTION
**Pull request by fisker**
## Description

1. Add `--minify` and `--no-minify` flags, for debug purpose
2. Change `--purge-cache` to `--no-cache`
3. Add readme, closes #11817

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/build/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+yarn build --print-size
+```
+
+### `--minify` and `--no-minify` flags
+
+Force minify files:
+
+```sh
+yarn build --minify
+```
```
Comment (sosukesuzuki) [inline scripts/build/README.md:41]: Our build script will minify by default, right? Do we need to make this explicit?

Comment (fisker) [reply scripts/build/README.md:41]: I think I'll use it with `--file` flag together, if I want debug build script.

Comment (sosukesuzuki) [reply scripts/build/README.md:41]: Okay, I have no better idea for this. But I think if this option is for debugging we should explicit it.

Comment (fisker) [reply scripts/build/README.md:45]: https://github.com/prettier/prettier/pull/11858/commits/f136b1e876e2fe0c8ff37df9d65f81b1fb8a8308

I'm going to need this to debug #11861

**Pull request merged**